cpp/include/nvstrings/NVText.h

```diff
@@ -143,6 +143,15 @@ class NVText
      */
     static unsigned int edit_distance( distance_type algo, NVStrings& strs1, NVStrings& strs2, unsigned int* results, bool devmem=true );
 
+    /**
+     * @brief Compute the edit distance between each pair of strings in given nvstrings object
+     * @param algo The edit distance algorithm to use for the computation.
+     * @param strs Strings to process.
+     * @param devmem True if results in device memory.
+     * @return 0 if successful.
+     */
+    static unsigned int edit_distance_matrix( distance_type algo, NVStrings& strs, unsigned int* results, bool devmem=true );
```

Comment:
I think the API design here is a bit problematic. The function sometimes returns its results in the pre-allocated device memory pointer `results`. However if `bdevmem` (unclear name) is false, it assume results is a host pointer, and allocates temporary memory and then copies the results back to the host. 

I think this complexity should be eliminated from the API. Instead, *always* return device memory, and if the caller needs the results on the host it can do the copy explicitly. This way there's no confusing API and no hidden copy.

Spell out names

```suggestion
    static unsigned int edit_distance_matrix( distance_type algorithm, NVStrings& strings, unsigned int* results, bool device_memory = true );
```

Reply:
Changing the way output `results` is returned is not straightforward. Since there is no cython binding layer in between cpp and python code. A decent amount of changes are needed in `cudf/python/nvstrings/cpp/pytext.cpp` which will not be relevant once  `PR-2811 cudf strings column` is merged.

But if you still think we should do it, let me know.